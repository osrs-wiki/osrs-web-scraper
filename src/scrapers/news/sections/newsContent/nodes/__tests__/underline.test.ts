import { MediaWikiBuilder } from "@osrs-wiki/mediawiki-builder";
import parse from "node-html-parser";

import boldParser from "../bold";

describe("underline node", () => {
  test("Underlined text should parse and render", () => {
    const root = parse("<u>test</u>");
    const builder = new MediaWikiBuilder();
    builder.addContents([boldParser(root.firstChild)].flat());
    expect(builder.build()).toMatchSnapshot();
  });
});
