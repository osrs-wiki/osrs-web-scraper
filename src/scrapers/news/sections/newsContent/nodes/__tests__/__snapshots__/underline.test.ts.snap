// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`underline node Underlined text should parse and render 1`] = `"'''test'''"`;

exports[`underline node Underlined text should parse and render with a link 1`] = `"'''testing this [https://example.com link]'''"`;

exports[`underline node Underlined text should parse and render with a link and bold 1`] = `"'''testing this [https://example.com link] and <!-- Unsupported tag: b -->'''"`;

exports[`underline node Underlined text should parse and render with a link and italics 1`] = `"'''testing this [https://example.com link] and ''italics'''''"`;
